src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestGetPolicyAction.kt

```diff
@@ -0,0 +1,93 @@
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.models.Policy.Companion.POLICY_TYPE
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.util._ID
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.util._VERSION
+import org.elasticsearch.action.get.GetRequest
+import org.elasticsearch.action.get.GetResponse
+import org.elasticsearch.client.node.NodeClient
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler
+import org.elasticsearch.common.xcontent.XContentHelper
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.rest.BaseRestHandler
+import org.elasticsearch.rest.BytesRestResponse
+import org.elasticsearch.rest.RestChannel
+import org.elasticsearch.rest.RestController
+import org.elasticsearch.rest.RestRequest
+import org.elasticsearch.rest.RestResponse
+import org.elasticsearch.rest.RestStatus
+import org.elasticsearch.rest.action.RestActions
+import org.elasticsearch.rest.action.RestResponseListener
+import org.elasticsearch.search.fetch.subphase.FetchSourceContext
+
+class RestGetPolicyAction(settings: Settings, controller: RestController) : BaseRestHandler(settings) {
+
+    init {
+        // Get a specific policy
+        controller.registerHandler(RestRequest.Method.GET, "$POLICY_BASE_URI/{policyID}", this)
+        controller.registerHandler(RestRequest.Method.HEAD, "$POLICY_BASE_URI/{policyID}", this)
+    }
+
+    override fun getName(): String {
+        return "get_policy_action"
+    }
+
+    override fun prepareRequest(request: RestRequest, client: NodeClient): RestChannelConsumer {
+        val policyId = request.param("policyID")
+        if (policyId == null || policyId.isEmpty()) {
+            throw IllegalArgumentException("Missing policy ID")
+        }
+        val getRequest = GetRequest(INDEX_STATE_MANAGEMENT_INDEX, INDEX_STATE_MANAGEMENT_TYPE, policyId)
+                .version(RestActions.parseVersion(request))
+
+        if (request.method() == RestRequest.Method.HEAD) {
+            getRequest.fetchSourceContext(FetchSourceContext.DO_NOT_FETCH_SOURCE)
+        }
+        return RestChannelConsumer { channel -> client.get(getRequest, getPolicyResponse(channel)) }
+    }
+
+    private fun getPolicyResponse(channel: RestChannel): RestResponseListener<GetResponse> {
+        return object : RestResponseListener<GetResponse>(channel) {
+            @Throws(Exception::class)
+            override fun buildResponse(response: GetResponse): RestResponse {
+                if (!response.isExists) {
+                    return BytesRestResponse(RestStatus.NOT_FOUND, channel.newBuilder())
+                }
+
+                val builder = channel.newBuilder()
+                        .startObject()
+                        .field(_ID, response.id)
+                        .field(_VERSION, response.version)
+                if (!response.isSourceEmpty) {
```

Comment:
NP: This `if` statement could be reformatted to a be a little more readable. Something like:
```
if (!response.isSourceEmpty) {
    XContentHelper.createParser(
        channel.request().xContentRegistry,
        LoggingDeprecationHandler.INSTANCE,
        response.sourceAsBytesRef,
        XContentType.JSON).use { xcp -> 
       	    val policy = Policy.parseWithType(xcp, response.id, response.version)
            builder.field(POLICY_TYPE, policy)
    	}
}
```

Reply:
I think would be good to use some standard formatter for editor whichever is convenient to a team.

Reply:
Changed